pdfgen/src/types/hierarchy/primitives/font.rs

```diff
@@ -0,0 +1,71 @@
+//! Implementation of PDF Font object.
+
+use crate::types::{self, constants};
+
+use super::{name::Name, obj_id::ObjId, object::Object};
+
+/// Represents a font object in a PDF document.
+/// This struct represents a font object in a PDF document, encapsulating the info required to
+/// define and reference a font, including its unique ID, subtype, and base font type.
+/// Fonts are essential for rendering text in PDFs and specify the appearance and
+/// characteristics of text elements.
+pub struct Font {
+    /// ID of this Font object.
+    id: ObjId,
+
+    /// Specifies the subtype of the font, defining its role or characteristics within the PDF.
+    subtype: Vec<u8>,
+
+    /// Represents the base font type, identifying the general font family or format.
+    base_type: Vec<u8>,
+}
+
+impl Font {
+    const FONT: Name<&'static [u8]> = Name::from_static(b"Font");
+    const SUBTYPE: Name<&'static [u8]> = Name::from_static(b"Subtype");
+    const BASE_FONT: Name<&'static [u8]> = Name::from_static(b"BaseFont");
+
+    /// Create a new Font object with the provided id, subtype and base_type.
+    pub fn new(id: ObjId, subtype: Vec<u8>, base_type: Vec<u8>) -> Self {
```

Comment:
We might for now add documentation where we explain that subtype and base_type need to be valid PDF `Name` types. If they aren't then we panic. We can also take these parameters as following: 

```suggestion
    pub fn new<S, B>(id: ObjId, subtype: S, base_type: B) -> Self 
    where
        S: AsRef<[u8]>,
        B: AsRef<[u8]>
{
```